src/components/utils/debounce.ts

```diff
@@ -0,0 +1,16 @@
+export function debounce<T extends (...args: any[]) => any>(
+  func: T,
+  wait: number,
+): (...args: Parameters<T>) => void {
+  let timeout: NodeJS.Timeout | null;
```

Comment:
```suggestion
  let timeout: ReturnType<typeof setTimeout> | null;
```

It's a bit more verbose but at least we don't leak NodeJS types in frontend code. The typing of setTimeout is annoying when `@types/node` is installed in a project...